tests/integration/security/authorization_test.go

```diff
@@ -1450,113 +1378,221 @@ extensionProviders:
-				newTestCase(x, g, "/custom", "http", "allow", true, expectHTTPResponse, scheme.HTTP),
-				newTestCase(x, g, "/custom", "http", "deny", false, expectHTTPResponse, scheme.HTTP),
-				newTestCase(x, g, "/health", "http", "allow", true, nil, scheme.HTTP),
-				newTestCase(x, g, "/health", "http", "deny", true, nil, scheme.HTTP),
+				newIngressTestCase(x, g, scheme.HTTP, "http", "/custom", authzHeaders("allow"), validateHTTPResponse, true),
+				newIngressTestCase(x, g, scheme.HTTP, "http", "/custom", authzHeaders("deny"), validateHTTPResponse, false),
+				newIngressTestCase(x, g, scheme.HTTP, "http", "/health", authzHeaders("allow"), nil, true),
+				newIngressTestCase(x, g, scheme.HTTP, "http", "/health", authzHeaders("deny"), nil, true),
 			}
-			for _, tc := range ingressCases {
-				name := fmt.Sprintf("%s->%s:%s%s[%t]",
-					tc.Request.From.Config().Service,
-					tc.Request.Options.Target.Config().Service,
-					tc.Request.Options.PortName,
-					tc.Request.Options.Path,
-					tc.ExpectAllowed)
-
-				t.NewSubTest(name).Run(func(t framework.TestContext) {
-					wantCode := map[bool]int{true: 200, false: 403}[tc.ExpectAllowed]
-					headers := map[string][]string{
-						"X-Ext-Authz": {tc.Headers["x-ext-authz"]},
-					}
-					authn.CheckIngressOrFail(t, ingr, "www.company.com", tc.Request.Options.Path, headers, "", wantCode)
-				})
+			for _, c := range ingressCases {
+				c(t)
 			}
 		})
 }
+
+type rbacTestName string
+
+func (n rbacTestName) String() string {
+	return string(n)
+}
+
+func (n rbacTestName) SkipIfNecessary(t framework.TestContext) {
+	t.Helper()
+
+	// Current source ip based authz test cases are not required in multicluster setup
+	// because cross-network traffic will lose the origin source ip info
+	if strings.Contains(n.String(), "source-ip") && t.Clusters().IsMulticluster() {
+		t.Skip("https://github.com/istio/istio/issues/37307")
+	}
+}
+
+func newRbacTestName(prefix string, expectAllowed bool, from echo.Instance, opts *echo.CallOptions) rbacTestName {
+	want := "deny"
+	if expectAllowed {
+		want = "allow"
+	}
+
+	return rbacTestName(fmt.Sprintf("%s%s->%s:%s%s[%s]",
+		prefix,
+		from.Config().Service,
+		opts.Target.Config().Service,
+		opts.PortName,
+		opts.Path,
+		want))
+}
+
+func expectRBACSuccessOrFailure(expectSuccess bool, from echo.Instance, to echo.Instances, opts *echo.CallOptions) echo.Validator {
+	if expectSuccess {
+		return expectRBACSuccess(from, to, opts)
+	}
+	return expectRBACFailure(from, to, opts)
+}
+
+func expectRBACSuccess(from echo.Instance, to echo.Instances, opts *echo.CallOptions) echo.Validator {
+	return echo.And(
+		echo.ValidatorFunc(func(resp client.ParsedResponses, err error) error {
+			if err == nil {
+				err = resp.CheckOK()
+			}
+			if err != nil {
+				return getError(from, opts, "allow with code 200", fmt.Sprintf("error: %v", err))
+			}
+			return nil
+		}),
+		validators.ExpectReachedClusters(to, opts))
```

Comment:
Not sure why yet, but I saw this test breaking for multi-network. If this PR doesn't somehow fix it: 

```suggestion
		validators.ExpectReachedClusters(to.Clusters().ByNetwork()[from.Config().Cluster.NetworkName()], opts))
```

Reply:
In a ~recent attempt to enable this for multi-cluster it was broken for multi-network.